### ISSUE TITLE
Update the minimum supported Node.js version

### ISSUE THREAD

**Issue by ericcornelissen** [ericcornelissen/shescape#193]
Update [the minimum supported Node.js version](https://github.com/ericcornelissen/shescape/blob/d1f31ccc612cc6bb883958cdec72b386af2894f6/package.json#L67) to v14 with the v2 major version bump. ~~As of writing this issue v12 is the oldest LTS Node.js release, but support for v12 ends 2022-04-30 ([[1]](https://nodejs.org/en/about/releases/)). Even if v2 of Shescape is released before that, it doesn't make much sense to provide support for it given the short support window (plus, v1 will be receiving security support for a while after the release of v2).~~ Neither Node.js v10 nor v12 are supported anymore.

**Comment by ericcornelissen** [comment]
Additionally, it should be considered to change from `>= x.x.x` to something like `^14.0.0 || ^16.0.0 || ^18.0.0`. This avoids the requirement to support odd-releases, which have a shorter LTS than even-releases.

Note, when doing this, adding a new supported Node.js version **isn't** a breaking change. That is, the following would not be breaking:

```diff
   "engines": {
-    "node": "^14.0.0 || ^16.0.0"
+    "node": "^14.0.0 || ^16.0.0 || ^18.0.0"
   }
```

For an example of a popular project that (roughly) takes this approach, see [Jest's `package.json`](https://github.com/facebook/jest/blob/3a8696bcf0e02a5f59080123d1a20e8b6d66d8df/packages/jest/package.json#L35).

**Comment by ericcornelissen** [comment]
This was first available in [2.0.0-rc1](https://www.npmjs.com/package/shescape/v/2.0.0-rc1).
